src/Machine.Specifications.Specs/Fixtures/RandomFixture.cs

```diff
@@ -1048,6 +1049,43 @@ public static void Reset()
             cleanup_invoked = false;
         }
     }
+
+    public class AsyncSpecifications
+    {
+        public static bool because_invoked;
+
+        public static bool async_it_invoked;
+
+        public static bool sync_it_invoked;
+
+        Because of = async () =>
+        {
+            because_invoked = true;
```

Comment:
This statement is executed synchronously with this setup, isn't it? It should be placed after the `await`, I guess.